src/hooks/useOnEscapePress.ts

```diff
@@ -1,23 +1,34 @@
-import {useEffect, useCallback} from 'react'
-
-const handlers: ((e: KeyboardEvent) => void)[] = []
+import {useEffect, useCallback, useMemo} from 'react'
 
 /**
  * Calls all handlers in reverse order
  * @param event The KeyboardEvent generated by the Escape keydown.
  */
 function handleEscape(event: KeyboardEvent) {
-  if (event.key === 'Escape' && !event.defaultPrevented) {
-    for (let i = handlers.length - 1; i >= 0; --i) {
-      handlers[i](event)
+  if (!event.defaultPrevented) {
+    for (const handler of Object.values(registry).reverse()) {
```

Comment:
Is the order of `Object.values(registry)` consistent/predictable?

Reply:
Would it make more sense to push and pop from a stack (array)?

Reply:
Yep! I found it more consistent than an array implementation I tried to build myself.